src/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -1143,6 +1145,17 @@ private String normalizeUrlSuffix(String url) {
         return url;
     }
 
+    private String stripIndexPhpOrAppsFiles(String url, EditText mHostUrlInput) {

```

Comment:
Can't this be achieved using URL class? It looks like it can be. Something like:
URL u = ...;
return u.toString().substring(0, u.toString().length() - u.getPath().length()-1);
?